_posts/2020-3-29-managed-services-serverless.md

```diff
@@ -0,0 +1,181 @@
+
+> Author bio: [Dan Mangum](https://twitter.com/hasheddan) is a software engineer at [Upbound](https://upbound.io/) where he works on the open source [Crossplane](https://crossplane.io/) project. He also serves on the Kubernetes release team, and is an active contributor to the Kubernetes project and multiple other open source efforts. He hosts a biweekly live stream show, [The Binding Status](https://www.youtube.com/playlist?list=PL510POnNVaaYFuK-B_SIUrpIonCtLVOzT), focused on extending Kubernetes, building Crossplane, and shaping the future of cloud-native computing.
+
+<img src="/images/2020-03-30-managed-services-serverless/mangum.jpg" width="25%" height="25%" />
+
+Kubernetes is often seen as complex and confusing for new and seasoned users alike. While it provides a wealth of tools and functionality for infrastructure operators, developers frequently struggle to navigate the complexity to just get their application running. OpenFaaS eases that burden by providing the fastest way to get your code running on Kubernetes.
+
+However, while OpenFaaS addresses the problem of getting your code to production, providing data for your functions is a different beast entirely. Most applications do not provide much value without data, so this is vital piece of the puzzle.
+
+# The Traditional Workflow
+
+Typically, the provisioning of managed services such as databases, queues, and caches follows a much different cycle than the deploying of an application. You must either use a cloud provider console, an infrastructure as code tool, or run the service directly on your Kubernetes cluster. Each of these methods have tradeoffs:
+
+- **Cloud Provider Console**: While probably the most developer-friendly route, using a user interface to deploy your managed services is tedious, not easily reproducible, and decoupled from the deployment of your functions on OpenFaaS.
+- **Infrastructure as Code / APIs**: Nearly every cloud provider supplies an API and set of SDKs that are used by infrastructure as code tools to make it possible to version control your infrastructure provisioning and configuration. This is a *massive* improvement over manual, ad-hoc provisioning, but typically requires a user executing command from their local machine, or manually editing detailed configuration every time they want a new service deployed. Once again, this process is also decoupled from deploying your functions on OpenFaaS.
+- **Running Data Services on Kubernetes Directly**: This path brings you closer to managing both your application code and data services with the same tools and workflow. However, it requires that you shun the high SLAs of experienced cloud providers in favor of administrating complex databases, caches, and queues yourself. This is generally a suitable option for large, experienced organizations, or applications that are not mission critical, but can impose significant burden on companies that are not in the business of running data services.
```

Comment:
I would like to see you mention other alternatives to Crossplane, such as Service Catalog via the Open Source Broker API

https://kubernetes.io/docs/concepts/extend-kubernetes/service-catalog/

Reply:
Hmm I would be happy to mention that the Kubernetes Service Catalog was an attempt to achieve similar functionality, but I don't think that it is very actively maintained at this point.

Reply:
I have made the other updates you have requested :+1:

Reply:
OSB Open Source Broker API is used extensively in CloudFoundry

Reply:
Not sure how to introduce this. It feels out of place to add something about CloudFoundry in this post that focuses on Kubernetes. Any thoughts?